backend/service/k8s/deployment.go

```diff
@@ -175,3 +212,30 @@ func mergeDeploymentLabelsAndAnnotations(deployment *appsv1.Deployment, fields *
 		deployment.Spec.Template.ObjectMeta.Annotations = labels.Merge(labels.Set(deployment.Spec.Template.ObjectMeta.Annotations), labels.Set(fields.Annotations))
 	}
 }
+
+func updateContainerResources(deployment *appsv1.Deployment, fields *k8sapiv1.UpdateDeploymentRequest_Fields) error {
+	for _, containerResource := range fields.ContainerResources {
+		for _, container := range deployment.Spec.Template.Spec.Containers {
+			if container.Name == containerResource.ContainerName {
+				resourceNames := []string{"cpu", "memory"}
```

Comment:
i noticed there's also a `hugepages` resource [1]. can we iterate over the passed in resources rather than these hardcoded resources names?

[1] https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-requests-and-limits-of-pod-and-container